File: Orimath.App/ViewModels/ViewPointConverter.fs
namespace Orimath.ViewModels
open Orimath.Core
open Orimath.Plugins

type ViewPointConverter(scale: float, offsetX: float, offsetY: float) =
    member __.ModelToView(point: Point) =
        ScreenPoint(point.X * scale + offsetX, point.Y * scale + offsetY)

    member __.ViewToModel(point: ScreenPoint) =
        { X = (point.X - offsetX) / scale; Y = (point.Y - offsetY) / scale }

    interface IViewPointConverter with
        member this.ModelToView(point) = this.ModelToView(point)
        member this.ViewToModel(point) = this.ViewToModel(point)

File: Orimath/ViewModels/ViewPointConverter.cs
using Orimath.Plugins;
using Point = Orimath.Core.Point;
using ViewPoint = System.Windows.Point;

namespace Orimath.ViewModels
{
    public class ViewPointConverter : IViewPointConverter
    {
        private readonly double _scaleX;
        private readonly double _scaleY;
        private readonly double _offsetX;
        private readonly double _offsetY;

        public ViewPointConverter(double scaleX, double scaleY, double offsetX, double offsetY)
        {
            _scaleX = scaleX;
            _scaleY = scaleY;
            _offsetX = offsetX;
            _offsetY = offsetY;
        }

        public ViewPoint ModelToView(Point point) => new ViewPoint(point.X * _scaleX + _offsetX, point.Y * _scaleY + _offsetY);

        public Point ViewToModel(ViewPoint point) => new Point((point.X - _offsetX) / _scaleX, (point.Y - _offsetY) / _scaleY);
    }
}
